### ISSUE TITLE
TypeScript definition

### ISSUE THREAD

**Issue by toverux** [bee-queue/arena#160]
Hello,

I rapidly wrote a small .d.ts file for Arena in order to be able to use it in my TypeScript project. I lack time to propose a PR or post it on DefinitelyTyped, so I'll just paste it here if that helps other people:

```ts
declare module 'bull-arena' {
    import { RequestHandler } from 'express';

    function BullArena(
        options: BullArena.MiddlewareOptions,
        listenOptions?: BullArena.MiddlewareListenOptions): RequestHandler;

    namespace BullArena {
        interface MiddlewareOptions {
            queues: Array<QueueOptions & ConnectionOptions>;
        }

        interface MiddlewareListenOptions {
            port?: number;
            host?: string;
            basePath?: string;
            disableListen?: boolean;
            useCdn?: boolean;
        }

        interface QueueOptions {
            name: string;
            hostId?: string;
            type?: 'bull' | 'bee';
            prefix?: 'bull' | 'bq' | string;
        }

        type ConnectionOptions = PortHostConnectionOptions | RedisUrlConnectionOptions | RedisClientConnectionOptions;

        interface PortHostConnectionOptions {
            host: string;
            port: number;
            password: string;
            db: string;
        }

        interface RedisUrlConnectionOptions {
            url: string;
        }

        interface RedisClientConnectionOptions {
            redis: object;
        }
    }

    export = BullArena;
}
```

**Comment by levibostian** [comment]
Thank you, @toverux! This has been very helpful!

For others who have found this issue: 
* [I made a PR into DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36210). If it gets merged, you will be able to use typings for arena by simply installing `@types/bull-arena`. 
* The typing definition above works for me, except I had to make 1 change:

Change:
```javascript
interface PortHostConnectionOptions {
    host: string;
    port: number;
    password: string;
    db: string;
}
```

to:

```javascript
interface PortHostConnectionOptions {
    host: string;
    port?: number;
    password?: string;
    db?: string;
}
```

Because these values are optional: 
![2019-06-14 at 4 34 PM](https://user-images.githubusercontent.com/2041082/59539343-78067700-8ec2-11e9-832f-4e97139bc376.png)
